tests/workflows/test_pytorch_optuna.py

```diff
@@ -0,0 +1,208 @@
+# Derived partially from https://pytorch.org/tutorials/beginner/dcgan_faces_tutorial.html
+import numpy as np
+import optuna
+import pytest
+import torch
+import torch.nn as nn
+import torch.nn.parallel
+import torch.optim as optim
+import torch.utils.data
+from dask.distributed import PipInstall, wait
+
+NC = 3  # Number of channels in the training images. For color images this is 3
+NZ = 100  # Size of z latent vector (i.e. size of generator input)
+
+
+# We create a fake image dataset. This ought to be replaced by
+# your actual dataset or pytorch's example datasets. We do this here
+# to focus more on computation than actual convergence.
+class FakeImageDataset(torch.utils.data.Dataset):
+    def __init__(self, count=1_000):
+        self.labels = np.random.randint(0, 5, size=count)
+
+    def __len__(self):
+        return len(self.labels)
+
+    def __getitem__(self, idx):
+        label = self.labels[idx]
+        torch.random.seed = label
+        # Spatial size of training images. All images will be resized to this
+        # size using a transformer.
+        img = torch.rand(3, 64, 64)
+        return img, label
+
+
+def weights_init(m):
+    if isinstance(m, (nn.ConvTranspose2d, nn.Conv2d)):
+        nn.init.normal_(m.weight.data, 0.0, 0.02)
+    elif isinstance(m, nn.BatchNorm2d):
+        nn.init.normal_(m.weight.data, 1.0, 0.02)
+        nn.init.constant_(m.bias.data, 0)
+
+
+def get_generator(trial):
+    # Create and randomly initialize generator model
+    activations = ["Sigmoid", "Tanh"]
+    activation = trial.suggest_categorical("generator_activation", activations)
+    ngf = 64  # Size of feature maps in generator
+    model = nn.Sequential(
+        # input is Z, going into a convolution
+        nn.ConvTranspose2d(NZ, ngf * 8, 4, 1, 0, bias=False),
+        nn.BatchNorm2d(ngf * 8),
+        nn.ReLU(True),
+        # state size. ``(ngf*8) x 4 x 4``
+        nn.ConvTranspose2d(ngf * 8, ngf * 4, 4, 2, 1, bias=False),
+        nn.BatchNorm2d(ngf * 4),
+        nn.ReLU(True),
+        # state size. ``(ngf*4) x 8 x 8``
+        nn.ConvTranspose2d(ngf * 4, ngf * 2, 4, 2, 1, bias=False),
+        nn.BatchNorm2d(ngf * 2),
+        nn.ReLU(True),
+        # state size. ``(ngf*2) x 16 x 16``
+        nn.ConvTranspose2d(ngf * 2, ngf, 4, 2, 1, bias=False),
+        nn.BatchNorm2d(ngf),
+        nn.ReLU(True),
+        # state size. ``(ngf) x 32 x 32``
+        nn.ConvTranspose2d(ngf, NC, 4, 2, 1, bias=False),
+        getattr(nn, activation)(),
+        # state size. ``(NC) x 64 x 64``
+    )
+    model.apply(weights_init)
+    return model
+
+
+def get_discriminator(trial):
+    # Create and randomly initialize discriminator model
+    activations = ["Sigmoid"]
+    activation = trial.suggest_categorical("discriminator_activation", activations)
```

Comment:
We're always selecting the `nn.Sigmoid` activation here -- are there others we could add?

Reply:
Added some more, and also to address the other comment about adding more hyperparameters in general.
https://github.com/coiled/benchmarks/pull/787/commits/55d8c1906d27c110734c637dd5a4594a789c79e5